pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go

```diff
@@ -0,0 +1,156 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	common_api "github.com/kumahq/kuma/api/common/v1alpha1"
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	core_xds "github.com/kumahq/kuma/pkg/core/xds"
+	api "github.com/kumahq/kuma/pkg/plugins/policies/meshhttproute/api/v1alpha1"
+	xds "github.com/kumahq/kuma/pkg/plugins/policies/meshhttproute/xds"
+	"github.com/kumahq/kuma/pkg/xds/envoy"
+	envoy_common "github.com/kumahq/kuma/pkg/xds/envoy"
+	envoy_listeners "github.com/kumahq/kuma/pkg/xds/envoy/listeners"
+	envoy_names "github.com/kumahq/kuma/pkg/xds/envoy/names"
+	envoy_tags "github.com/kumahq/kuma/pkg/xds/envoy/tags"
+	"github.com/kumahq/kuma/pkg/xds/generator"
+)
+
+func generateListeners(
+	proxy *core_xds.Proxy,
+	rules []ToRouteRule,
+	servicesAcc envoy_common.ServicesAccumulator,
+) (*core_xds.ResourceSet, error) {
+	resources := core_xds.NewResourceSet()
+	splitCounter := &splitCounter{}
+
+	for _, outbound := range proxy.Dataplane.Spec.GetNetworking().GetOutbound() {
+		serviceName := outbound.GetTagsIncludingLegacy()[mesh_proto.ServiceTag]
+		oface := proxy.Dataplane.Spec.Networking.ToOutboundInterface(outbound)
+		outboundListenerName := envoy_names.GetOutboundListenerName(oface.DataplaneIP, oface.DataplanePort)
+
+		listenerBuilder := envoy_listeners.NewListenerBuilder(proxy.APIVersion).
+			Configure(envoy_listeners.OutboundListener(outboundListenerName, oface.DataplaneIP, oface.DataplanePort, core_xds.SocketAddressProtocolTCP)).
+			Configure(envoy_listeners.TransparentProxying(proxy.Dataplane.Spec.Networking.GetTransparentProxying())).
+			Configure(envoy_listeners.TagsMetadata(envoy_tags.Tags(outbound.GetTagsIncludingLegacy()).WithoutTags(mesh_proto.MeshTag)))
+
+		for _, route := range FindRoutes(rules, serviceName) {
```

Comment:
Yeah, I think this was a bad refactoring of the configurers

Reply:
fixed

Reply:
Shouldn't this `FindRoutes` be moved inside `MatchedPolicies` and `TypedMatchingPolicies` be extended with a `ToRouteRules` ? This would make extending the inspect API easier in the future.

Reply:
Hold on, the result of `FindRoutes` depends on which outbound we're talking about.
Right now `ToRules` contains an "unmerged" set of `RuleConfs`, but we can't merge it until we know which outbound we're talking about. This is the same in other plugins, in that we call `ToRules.Rules.Compute` for each outbound. Does the inspect API support seeing the rules for a specific outbound yet?

Reply:
> This is indeed better for the future inspect API because the existing approach doesn't allow you to see the resulting RuleConf in TypeMatchingPolicies.

But I don't see why we should create a `ToRouteRules`, the problem is more that we can't put the merged rules into `ToRules`, isn't it?

Reply:
I'd much rather make `ToRules` more flexible than add a special case

Reply:
@lahabana @lobkovilya Can we tackle the inspect API in a separate PR? It's not broken ATM and this one is pretty big already

Reply:
Ok as long as it's a fast follow :)